scripts/quick_install.sh

```diff
@@ -0,0 +1,230 @@
+helm repo add hashicorp https://helm.releases.hashicorp.com
+helm install vault hashicorp/vault --namespace=vault --version=${VAULT_VERSION} --values=vault-values.yaml --wait --timeout=600s
+sleep 5
+kubectl wait pod/vault-0 --namespace=vault --for=condition=ready --timeout=600s
+
+# Downgrade to Vault KV secrets engine version 1
+kubectl exec vault-0 --namespace=vault -- vault secrets disable secret
+kubectl exec vault-0 --namespace=vault -- vault secrets enable -version=1 -path=secret kv
+
+# Write cluster credential to be saved in Vault
+cat <<EOF > cluster-credential.json
+{
+  "name": "dev",
+  "master_ip": "https://kubernetes.default.svc:443",
+  "certs": "$(cat ~/.minikube/ca.crt | awk '{printf "%s\\n", $0}')",
+  "client_certificate": "$(cat ~/.minikube/profiles/minikube/client.crt | awk '{printf "%s\\n", $0}'))",
+  "client_key": "$(cat ~/.minikube/profiles/minikube/client.key | awk '{printf "%s\\n", $0}'))"
+}
+EOF
+
+kubectl cp cluster-credential.json vault/vault-0:/tmp/cluster-credential.json
+kubectl exec vault-0 --namespace=vault -- vault kv put secret/${CLUSTER_NAME} @/tmp/cluster-credential.json
+
+# Install Minio
+cat <<EOF > minio-values.yaml
+replicas: 1
+persistence:
+  enabled: false
+resources:
+  requests:
+    cpu: 25m
+    memory: 64Mi
+livenessProbe:
+  initialDelaySeconds: 30
+defaultBucket:
+  enabled: true
+  name: mlflow
+ingress:
+  enabled: true
+  annotations:
+    kubernetes.io/ingress.class: istio
+  path: /*
+  hosts:
+    - 'minio.minio.${INGRESS_HOST}.nip.io'
+EOF
+
+kubectl create namespace minio
+helm repo add minio https://helm.min.io/
+helm install minio minio/minio --version=${MINIO_VERSION} --namespace=minio --values=minio-values.yaml --wait --timeout=600s
+
+# Install MLP
+kubectl create namespace mlp
+
+git clone git@github.com:gojek/mlp.git
+
+helm install mlp ./mlp/chart --namespace=mlp --values=./mlp/chart/values-e2e.yaml \
+  --set mlp.image.tag=main \
+  --set mlp.apiHost=http://mlp.mlp.${INGRESS_HOST}.nip.io/v1 \
+  --set mlp.oauthClientID=${OAUTH_CLIENT_ID} \
+  --set mlp.mlflowTrackingUrl=http://mlflow.mlp.${INGRESS_HOST}.nip.io \
+  --set mlp.ingress.enabled=true \
+  --set mlp.ingress.class=istio \
+  --set mlp.ingress.host=mlp.mlp.${INGRESS_HOST}.nip.io \
+  --set mlp.ingress.path="/*" \
+  --wait --timeout=5m
+
+cat <<EOF > mlp-ingress.yaml
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: mlp
+  namespace: mlp
+  annotations:
+    kubernetes.io/ingress.class: istio
+  labels:
+    app: mlp
+spec:
+  rules:
+    - host: 'mlp.mlp.${INGRESS_HOST}.nip.io'
+      http:
+        paths:
+          - path: /*
+            backend:
+              serviceName: mlp
+              servicePort: 8080
+EOF
+
+# Install Merlin
+kubectl create secret generic vault-secret --namespace=mlp --from-literal=address=http://vault.vault.svc.cluster.local --from-literal=token=root
+
+helm install merlin ../charts/merlin --namespace=mlp --values=../charts/merlin/values-e2e.yaml \
```

Comment:
I followed the same steps to try it out on minikube with virtual box. I run into below error `/src/api/cmd/main.go:126 +0x7f7
{"level":"panic","ts":1610364390.0461166,"caller":"cmd/main.go:554","msg":"unable to retrieve secret for cluster dev from vault","stacktrace":"github.com/gojek/merlin/log.Panicf\n\t/src/api/log/log.go:56\nmain.initImageBuilder\n\t/src/api/cmd/main.go:554\nmain.main\n\t/src/api/cmd/main.go:126\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203"}
panic: unable to retrieve secret for cluster dev from vault`. All other pods are in running state except this. Vault pod is running and I am able to a get the secret using the command `kubectl exec vault-0 --namespace=vault -- vault kv get secret/${CLUSTER_NAME}`. I also tried setting external vault but no luck. Any help is appreciated.

Reply:
What's the output of this command: `export CLUSTER_NAME=dev && kubectl exec vault-0 --namespace=vault -- vault kv get secret/${CLUSTER_NAME}`?
It's expected that the output contains the Kubernetes cluster credentials

Otherwise, it's possible that the script fails to write the credential to vault (https://github.com/gojek/merlin/blob/local-dev-doc/scripts/quick_install.sh#L131-L147).

Reply:
the command outputs the credentials.

Reply:
Should the address in the below command have port as well? 
`kubectl create secret generic vault-secret --namespace=mlp --from-literal=address=http://vault.vault.svc.cluster.local --from-literal=token=root`

Reply:
> Should the address in the below command have port as well?
No need. Merlin can connect to Vault.

Can you please re-run this part of scripts: https://github.com/gojek/merlin/blob/local-dev-doc/scripts/quick_install.sh#L131-L147 ?
Seems like there's a discrepancy on Vault secret engine that Merlin needs. After that, please verify if secret exist then try to redeploy Merlin pod (or reinstall Merlin helm).

Reply:
Trying that. I will let you know in a while.

Reply:
It works after re-deployment. Not sure what I had missed in the 1st step.

Reply:
Nice!

Seems like the vault isn't ready yet when we run these (https://github.com/gojek/merlin/blob/local-dev-doc/scripts/quick_install.sh#L132-L133)